library/ecp.c

```diff
@@ -767,6 +767,63 @@ int mbedtls_ecp_point_write_binary( const mbedtls_ecp_group *grp,
     return( ret );
 }
 
+#if defined(MBEDTLS_ECP_SHORT_WEIERSTRASS_ENABLED)
+static int mbedtls_ecp_sw_derive_y( const mbedtls_ecp_group *grp,
```

Comment:
(Making a thread to keep the discussion grouped)

[@mpg writes](https://github.com/Mbed-TLS/mbedtls/pull/6282#pullrequestreview-1115222466)

> Perhaps we'll want this to be guarded by a compile-time option as well.

[and further](https://github.com/Mbed-TLS/mbedtls/pull/6282#issuecomment-1257862273)

> > Agreed. What do you suggest for a compile-time option name?
>
> I don't know. Perhaps MBEDTLS_ECP_READ_COMPRESSED_FORMAT (to highlight that we always support writing this format).

Looking at the size of the code, I don't think it warrants a compile-time option. It's just adding about 60 lines to `ecp*.c` which is ~9000 lines. Each compile-time option complicates maintenance and increases the complexity of testing. That's not worth it here.

Reply:
This function is about 280 bytes on Arm v6-M. It's indeed not much compared to the 9695 bytes of `ecp.o`.

OTOH, when working against a tight target in the baremetal branch, we made quite a few optimizations that were smaller than that and if you add all the lines < 280 from [this table](https://github.com/Mbed-TLS/mbedtls/issues/3535#issuecomment-683674260) it adds up 1964 which can be significant in some cases.

That said, these days I think too many options is more of a problem than 280 bytes, and when it comes to code size we have a lot of bigger things to do before worrying about 280 here and there becomes meaningful, so I'm OK with making it non-optional.